nodes/exchange/FCStd_write.py

```diff
@@ -0,0 +1,182 @@
+
+from sverchok.dependencies import FreeCAD
+from sverchok.utils.dummy_nodes import add_dummy
+
+if FreeCAD is None:
+    add_dummy('SvWriteFCStdNode', 'SvWriteFCStdNode', 'FreeCAD')
+
+else:
+    F = FreeCAD
+    import bpy,sys
+    from bpy.props import IntProperty, FloatProperty, StringProperty, BoolProperty,EnumProperty
+    from sverchok.node_tree import SverchCustomTreeNode, throttled
+    from sverchok.data_structure import updateNode
+    from numpy import ndarray
+
+    class SvWriteFCStdNode(bpy.types.Node, SverchCustomTreeNode):
+        ''' SvWriteFCStdNode '''
+        bl_idname = 'SvWriteFCStdNode'
+        bl_label = 'Write FCStd'
+        bl_icon = 'IMPORT'
+        solid_catergory = "Inputs"
+        
+        write_update : BoolProperty(
+            name="write_update", 
+            default=True)
+
+        part_name : StringProperty(
+            name="part_name", 
+            default="part_name")
+
+        @throttled
+        def changeMode(self, context):
+
+            if self.obj_format == 'mesh':
+                if 'Verts' not in self.inputs:
+                    self.inputs.remove(self.inputs['Solid'])
+                    self.inputs.new('SvVerticesSocket', 'Verts')
+                    self.inputs.new('SvVerticesSocket', 'Faces')
+                    return
+            else:
+                if 'Solid' not in self.inputs:
+                    self.inputs.remove(self.inputs['Verts'])
+                    self.inputs.remove(self.inputs['Faces'])
+                    self.inputs.new('SvSolidSocket', 'Solid')
+                    return
+
+        
+        obj_format : EnumProperty(
+                    name='format',
+                    description='choose format',
+                    items={
+                    ('solid', 'solid', 'solid'),
+                    ('mesh', 'mesh', 'mesh')},
+                    default='solid',
+                    update=changeMode)
+
+        def draw_buttons(self, context, layout):
+
+            layout.label(text="write name:")
+            col = layout.column(align=True)
+            col.prop(self, 'part_name',text="")  
+            col.prop(self, 'obj_format',text="")     
+            col.prop(self, 'write_update')
+            if self.obj_format == 'mesh':
+                col.label(text="need triangle meshes")
+
+
+        def sv_init(self, context):
+            self.inputs.new('SvFilePathSocket', "File Path")
+
+            if self.obj_format == 'mesh':
+                self.inputs.new('SvVerticesSocket', "Verts")
+                self.inputs.new('SvStringsSocket', "Faces")
+
+            else:
+                self.inputs.new('SvSolidSocket', 'Solid')
+           

```

Comment:
When this function is called obj_format will always be the default value (solid)

Reply:
Yes,  I noticed a strange behaviour with the enum, also if I save somethime when I reopen the file the value is changed. 
I'll look better at how this is implemented in other nodes.